src/pkg/packager/create.go

```diff
@@ -208,9 +208,10 @@ func (p *Packager) pullImages(imgList []string, path string) (map[name.Tag]v1.Im
 
 	return pulledImages, utils.Retry(func() error {
 		imgConfig := images.ImgConfig{
-			TarballPath: path,
-			ImgList:     imgList,
-			Insecure:    p.cfg.CreateOpts.Insecure,
+			TarballPath:  path,
+			ImgList:      imgList,
+			InsecurePull: p.cfg.CreateOpts.Insecure,
+			InsecurePush: p.cfg.InitOpts.Insecure || p.cfg.DeployOpts.Insecure,
```

Comment:
@YrrepNoj @Racer159 I wonder if we should consider changing to an overall `insecure` flag in `p.cfg.CommonOpts` and the root cmd vs all these random `insecure` flags we have floating around for different parts of Zarf?

Reply:
I would agree with that.  It would simplify the logic and create (pull) and init/deploy (push) are already separate commands

Reply:
@jeff-mccoy, @YrrepNoj, @Racer159 Sure.. will do these changes in this PR itself.

Reply:
Changes are done in https://github.com/defenseunicorns/zarf/pull/1191/commits/ba3281bba2b85d66b1d322cdb5c6c02edc449f8c